android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt

```diff
@@ -274,11 +298,112 @@ open class GestureHandler<ConcreteGestureHandlerT : GestureHandler<ConcreteGestu
     }
   }
 
+  fun updatePointerData(event: MotionEvent) {
+    trackedPointers?.let { trackedPointers ->
+      pointerEventPayload = null
+
+      val offsetX = event.rawX - event.x
+      val offsetY = event.rawY - event.y
+
+      if (event.actionMasked == MotionEvent.ACTION_DOWN || event.actionMasked == MotionEvent.ACTION_POINTER_DOWN) {
+        pointerEventType = RNGestureHandlerPointerEvent.EVENT_POINTER_DOWN
+        val pointerId = event.getPointerId(event.actionIndex)
+
+        trackedPointers[pointerId] = PointerData(
+            pointerId,
+            event.getX(event.actionIndex),
+            event.getY(event.actionIndex),
+            event.getX(event.actionIndex) + offsetX,
+            event.getY(event.actionIndex) + offsetY,
+        )
+        trackedPointersCount++
```

Comment:
Do we really need separate counter for this? Maybe we can use trackedPointersIDCount, which I believe should always be in sync. Unless we expect pointer tracking to be enabled while gesture is already running, which does not make sense in my opinion. I think we can also just use a fixed-size array of `PointerData` for `trackedPointers`  and only populate entries which have active pointers + when generating event we'd ignore `null` entries

Reply:
There is one particular moment when they are not in sync (by design): currently `numberOfPointers` in pointer events represents number of pointers on the screen, so if you have one pointer on the screen and you lift it, you would get `UP` event with `numberOfPointers` set to zero. Should that be changed?

Reply:
As for the fixed array, done in https://github.com/software-mansion/react-native-gesture-handler/pull/1673/commits/e6ea1e68af716c619a34f032406b74219b9baad5 and https://github.com/software-mansion/react-native-gesture-handler/pull/1673/commits/fef255c2e23b9e795ad52bae965a1f30a65e387d.

Reply:
As for the UP event with numberOfPointers = 0 – it I think this is fine so long it works consistently. It is quite and edge case, as when there are no pointers and you send numberOfPointers = 1 one would expect the pointers data to also have an entry, but then what coordinates you put there? the pointer is not on screen so it has no position?